heartbeat.go

```diff
@@ -16,49 +16,40 @@ import (
 
 const (
 	// One hour interval should be good enough to renew tokens for four hours master token validity
-	heartBeatInterval = 60 * time.Minute
+	heartBeatInterval = 3600 * time.Second
 )
 
 type heartbeat struct {
 	restful      *snowflakeRestful
-	ShutdownChan chan int
+	shutdownChan chan bool
 }
 
 func (hc *heartbeat) run() {
-	hbTimer := time.NewTimer(heartBeatInterval)
+	hbTicker := time.NewTicker(heartBeatInterval)
+	defer hbTicker.Stop()
 	for {
-		go func() {
```

Comment:
This was causing the go-routines to be started per each call to exec, right?  So we will still have a timer running per exec call until Close() gets called on the connection?

Reply:
Instead of using Timer, Ticker is used to run heartbeat hourly. The difference is before this goroutine checks the shutdown signal *every second*, and the new code uses `select` statement to wait for either 1) shutdown signal or 2) trigger heartbeat event.